### ISSUE TITLE
Settings reset to default when updating

### ISSUE THREAD

**Issue by AyrA** [Thibaut/devdocs#662]
Please stop resetting all settings to default and delete the documentation cache each time devdocs is updated.

After every update that shows the reload notification on the top right all settings are lost and the documentation cache is erased. This happens on all devices I own.

**Comment by Thibaut** [comment]
Settings are not supposed to be reset when DevDocs is updated.

Which browser are you using? Do you have any extensions/addons installed?

**Comment by AyrA** [comment]
I am using chrome. The only extension installed that I can imagine could interfere with it is uBlock origin. From its logs it does not looks like it is blocking anything from the devdocs.io domain.

The behavior is not constant. I have two computers, one where chrome is permanently open and one where it is not. Only the one that has daily chrome shutdowns is affected by this issue.

**Comment by AyrA** [comment]
An alternative would be to allow users to export settings in a format like this:

```json
{
	"docs":[
		"apache_http_server",
		"bootstrap~3",
		"c", "cpp", "css",
		"dom", "dom_events",
		"git", "html", "http",
		"javascript", "jquery",
		"markdown", "moment",
		"node~6_lts", "php",
		"sqlite"
	],
	"settings":["dark"]
}
```

I could then re-import the settings once they are reset. This would be much less painful than selecting all documentations I need over and over again. It's also probably easy to implement.

I experimented and the problem is when the cookie gets deleted. I can simply add the "docs" key back to the cookie and then all documentations are selected again.

**Comment by Thibaut** [comment]
> An alternative would be to allow users to export settings in a format like this:

While that's an interesting idea that could be useful for other use-cases, I'd rather get to the bottom of this issue, rather than implement a work-around that doesn't address the root cause.

> Only the one that has daily chrome shutdowns is affected by this issue.

Do you get similar issues (e.g. being logged out) on other sites?

Could you post a screenshot of your cookies settings (`chrome://settings/content/cookies`) please.

**Comment by AyrA** [comment]
> Do you get similar issues (e.g. being logged out) on other sites?

This is not an issue for me, I actually want this to happen

> Could you post a screenshot of your cookies settings

Sure: https://i.imgur.com/GEqcX7F.png

**Comment by Thibaut** [comment]
Ok so this isn't an issue in DevDocs, since your browser is set to reset all data when it quits (the fact that DevDocs is reset when its application cache is updated could be a browser bug, but is definitely not a bug in DevDocs).

> By the way: `JSON.stringify(localStorage).length` reports over 2MB which puts this site very high into the list of being removed by the browser since localStorage is limited to 5MB on most. 

DevDocs uses `localStorage` only as a cache. If `localStorage` is cleared or unavailable, the app will continue to work just fine, including offline mode/data.

> An alternative would be to allow users to export settings in a format like this:

I've addressed this as part of #671.

**Comment by TuringTux** [comment]
The same symptoms came back (although, I believe, for different reasons, namely cookie expiry dates) 6–8 years later in #1765 (and #2037).
